### DISCUSSION TITLE
unhandled error in lsp from rpc.lua

### DISCUSSION THREAD

**Discussion by pannet1** [LazyVim/LazyVim#1134] [Q&A] [answered]
Hi,
i get a warning 
```[lspconfig] unhandled error: /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:663: cmd: expected string, got nil```
this is not a deal breaker. just worried if this has something to do with my setup.

**Comment by dpetka2001** [comment]
Maybe some misconfigured Lsp server? `cmd: expected string, got nil` makes me think that the `cmd` parameter of some lsp server is not properly configured. Do you have a link to Github for your Lazyvim configuration?

**Reply by pannet1** [reply]
[here](https://github.com/pannet1/config) did not touch the LazyVim factory configuration.

**Reply by folke** [reply, marked as answer]
Those dots are not public, so we can't see them.

Either way, this is exactly what the error says. The lsp server's cmd should be a string or function, but it seems it is not set in your config

**Reply by dpetka2001** [reply]
If this [repo](https://github.com/pannet1/LazyVim/blob/main/lua/plugins/lsp.lua) is your current configuration, then you're setting the `cmd` for both `pyright` and `jsonls` to empty table. Like @folke mentioned `cmd` should be a string or function. Since you're not setting anything, you could just omit the `cmd` line and `lspconfig` will pick up from default server configuration.

**Reply by pannet1** [reply]
@folke sorry, my bad, pointed to the wrong repo. [this](https://github.com/pannet1/LazyVim/) is the correct one. i understood what is causing the issue.

**Reply by pannet1** [reply]
@dpetka2001  thank you my friend Iordanis. got misguided by the line numbers in the error message. fixing and committing the repo now.
